internal/mode/static/state/resolver/resolver.go

```diff
@@ -149,7 +152,7 @@ func getDefaultPort(svcPort v1.ServicePort) int32 {
 }
 
 func ignoreEndpointSlice(endpointSlice discoveryV1.EndpointSlice, port v1.ServicePort) bool {
-	if endpointSlice.AddressType != discoveryV1.AddressTypeIPv4 {
+	if endpointSlice.AddressType == discoveryV1.AddressTypeFQDN {
```

Comment:
we should do this conditionally based on configured IP family:
- if IPv4, we should ignore IPv6 And AddressTypeFQDN)
- if IPv6 - IPv4 and AddressTypeFQDN
- Dual - ignore AddressTypeFQDN

sorry, missed this during previous review

Reply:
If a service is created with 

1. `Dual` stack it enables IPv4 and IPv6.
2. `IPv4` stack it enables only IPv4.
3. `IPv6` stack it enabled only IPv6.

If a service tries to create `IPv6` endpoints, on a cluster that only allows IPv4, it throws an error before allowing it to configure. So, that check is already done. 

For example: 
```
gateway.gateway.networking.k8s.io/gateway created
The Service "coffee" is invalid: spec.ipFamilies[0]: Invalid value: "IPv4": not configured on this cluster
```

Is this what you expect? Based on the service IPFamily, it already configures the correct IPs for it

Reply:
we have a discrepancy here.

if we configure IPFamily in NginxProxy to IPv4, NGF will configure listen socket to be IPv4, but will still allow Endpoints that are IPv6. Basically, we rely on NginxProxy to configure listen socket, but have no control over Endpoints

consider two cases:
- Kubernetes: dual case, NginxProxy: IPv4 only. Kubernetes Endpoints can still be either IPv4 or IPv6. If their are IPv6, let's ignore them and respect NginxProxy configuration.
- Kubernetes: dual case, NginxProxy: IPv6 only. Kubernetes Endpoints can still be either IPv4 or IPv6. If their are IPv4, let's ignore them and respect NginxProxy configuration.

Reply:
I have updated the resolver to filter out endpoints based on the IP Family configured for NGINX.